File: mvp/sdk/python/saas_ia/api/sentiment.py
"""Sentiment API — text sentiment analysis (RoBERTa + LLM)."""

from __future__ import annotations

from typing import Any

from saas_ia.api.base import BaseAPI


class SentimentAPI(BaseAPI):
    """Wraps ``/api/sentiment`` endpoints."""

    async def analyze(self, text: str) -> dict[str, Any]:
        """Analyze text sentiment and emotions."""
        return await self._post("/api/sentiment/analyze", json={"text": text})

    async def analyze_transcription(
        self, transcription_id: str
    ) -> dict[str, Any]:
        """Analyze sentiment of a transcription."""
        return await self._post(
            "/api/sentiment/transcription",
            json={"transcription_id": transcription_id},
        )

File: mvp/sdk/typescript/src/api/sentiment.ts
import { BaseAPI } from "./base";
import {
  SentimentAnalyzeRequest,
  SentimentResult,
  SentimentTranscriptionRequest,
} from "../types";

/**
 * Sentiment API — text sentiment analysis using RoBERTa + LLM fallback.
 */
export class SentimentAPI extends BaseAPI {
  /** Analyze text sentiment and emotions. */
  async analyze(data: SentimentAnalyzeRequest): Promise<SentimentResult> {
    return this._post<SentimentResult>("/api/sentiment/analyze", data);
  }

  /** Analyze sentiment of a transcription. */
  async analyzeTranscription(data: SentimentTranscriptionRequest): Promise<SentimentResult> {
    return this._post<SentimentResult>("/api/sentiment/transcription", data);
  }
}
